flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionManagerTest.java

```diff
@@ -60,4 +61,26 @@ public void testCreateViewForRegisteredPartition() throws Exception {
 		partitionManager.registerResultPartition(partition);
 		partitionManager.createSubpartitionView(partition.getPartitionId(), 0, new NoOpBufferAvailablityListener());
 	}
+
+	@Test
+	public void testExternallyManagedPartitionReleaseIfFlagDisabled() {
+		final ResultPartitionManager partitionManager = new ResultPartitionManager(false);
+		final ResultPartition partition = new ResultPartitionBuilder().setIsExternallyManaged(true).build();
+
+		partitionManager.registerResultPartition(partition);
+		partitionManager.onConsumedPartition(partition);
+
+		assertThat(partition.isReleased(), is(false));
+	}
+
+	@Test
+	public void testExternallyManagedPartitionReleaseIfFlagEnabled() {
```

Comment:
These two tests could be deduplicated by extracting a boolean value if you like.